### PR TITLE
Fix secondary cidr creation on the first pass

### PR DESCRIPTION
**Pull request by alexohima**
---

**Description:**
This pull request addresses a bug with the secondary CIDR creation in the module. The `create_vpc` local is fairly complex, taking into account whether `vpc_id` is passed or not. When setting up a secondary CIDR, we pass in `vpc_id` from another resource that hasn't been built yet, leading to a race condition and application errors. To fix this, we can introduce a `create_vpc` variable to replace the local. This fix ensures secondary CIDR creation within the first Terraform apply.

**Changes Made:**
- Updated the condition for creating a VPC to utilize a variable instead of relying on the VPC's ID.
- Added the variable `azs` to enable the calculation of AZs outside the module. During testing, we encountered an issue with creating a secondary CIDR where `local.private_per_az` and `local.azs` would be unknown until after the apply. The problem lies in the calculation of `local.azs` within the module. By passing it as `var.azs` instead, we resolve this issue.
- Defaulted `az_count` to 0 making it optional in the case we are already using `azs`.
- Provided an example demonstrating how to implement this update in the secondary CIDR example.

**Related Issues:**
https://github.com/aws-ia/terraform-aws-vpc/issues/142

---

### CODE REVIEW

**Comment by alexohima** [comment]
@drewmullen Can we get an update on this?

**Comment by Pacobart** [comment]
@tonynv @andrew-glenn @tlindsay42 We are waiting on this change to go in. Its necessary in order to build EKS subnets using secondary cidrs. Please review and approve.

File: data.tf [hunk 1]
```diff
@@ -89,8 +89,7 @@ locals {
   # - get cidr block value from AWS IPAM
   # - create flow logs
 
-  # # if var.vpc_id is passed, assume create = `false` and cidr comes from data.aws_vpc
-  create_vpc = var.vpc_id == null ? true : false
+  create_vpc = var.create_vpc ? true : false
```
Comment (drewmullen) [inline data.tf:92]: You shouldn’t need a bool local that’s just a ternary bool 

just replace local.create_vpc with the variable

Comment (alexohima) [reply data.tf]: That's true, I've replaced it.

Comment (Pacobart) [reply data.tf]: I've been using this branch to successfully build around 10 vpc's so far. Code changes are good. @drewmullen are we ready to merge?

**Review by Pacobart — APPROVED** [review]
I've evaluated these changes and deployed ~10 vpc's successfully with this logic. It works as expected.

Here's my code:
```
module "vpc" {
  source  = "aws-ia/vpc/aws"
  version = "4.4.2"

  name = local.name_with_region
  az_count = var.availability_zone_count
}

module "eks_secondary" {
  source = "git::https://github.com/alexohima/terraform-aws-ia-vpc?ref=fix-secondary-cidr-first-pass" # TEMP needed until workaround merged into main https://github.com/aws-ia/terraform-aws-vpc/pull/151
  #source     = "aws-ia/vpc/aws"
  #version    = "4.4.2"
  count = var.enable_eks_subnets ? 1 : 0
  name  = local.name_with_region
  azs        = module.vpc.azs
  cidr_block = "100.64.0.0/16"
  create_vpc = false # NOTE: available on feature branch only right now

  vpc_secondary_cidr = true
  vpc_id             = local.vpc_id

  subnets = {
    eks = {
      name_prefix = "${local.name}-eks"
      netmask     = 18
    }
  }
  depends_on = [module.vpc]
}
```

**Comment by drewmullen** [comment]
Any chance you can try your initial issue using the new alpha build with the deferred actions experiment enabled and then `terraform apply -allow-deferral`? I would like to avoid this change if possible

**Comment by alexohima** [comment]
This is unachievable for any organization currently using Control Tower Account Factory for Terraform without the considerable effort of forking and maintaining the entire [AFT repository](https://github.com/aws-ia/terraform-aws-control_tower_account_factory/blob/aee036db9d4926d60fd48a04fbc1d70917749456/modules/aft-customizations/buildspecs/aft-account-customizations-terraform.yml#L118)

**Comment by greenflowers** [comment]
@tonynv @andrew-glenn @tlindsay42 - I would also be keen on this feature!

**Comment by Pacobart** [comment]
@drewmullen anything else you need before we can get this merged?

**Review by drewmullen — APPROVED** [review]
LGTM

**Comment by Pacobart** [comment]
@tonynv @andrew-glenn @tlindsay42 please approve and merge in. We've been waiting for this to go through for a few weeks now.

**Review by tbulding — APPROVED** [review]

**Comment by Pacobart** [comment]
@tbulding now that this is merged can we get a new release tag create please?

**Comment by Pacobart** [comment]
@tonynv  @andrew-glenn  @tlindsay42  @drewmullen  - Can we get a new git release tag generated for 4.4.3 with these changes please?

**Comment by Drewster727** [comment]
Not a fan of targeting `main` branch. Any word on cutting a new release for this?

**Comment by andrew-glenn** [comment]
I'm just now seeing this. Apologies for the delay. A new release is imminent.

**Comment by tlindsay42** [comment]
+1. Done deal. Sorry for the delay. https://github.com/aws-ia/terraform-aws-vpc/releases/tag/v4.4.3

**Pull request merged**